specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/stable/2022-10-01/deviceupdate.json

```diff
@@ -0,0 +1,4472 @@
+      "x-ms-parameter-grouping": {
+        "name": "account-options"
+      }
+    },
+    "InstanceIdParameter": {
+      "name": "instanceId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "description": "Account instance identifier.",
+      "x-ms-parameter-location": "client",
+      "x-ms-parameter-grouping": {
+        "name": "account-options"
+      }
+    },
+    "UpdateProviderParameter": {
+      "name": "provider",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Update provider.",
+      "x-ms-parameter-location": "method"
+    },
+    "UpdateNameParameter": {
+      "name": "name",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Update name.",
+      "x-ms-parameter-location": "method"
+    },
+    "UpdateVersionParameter": {
+      "name": "version",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Update version.",
+      "x-ms-parameter-location": "method"
+    },
+    "FileIdParameter": {
+      "name": "fileId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "File identifier.",
+      "x-ms-parameter-location": "method"
+    },
+    "IfNoneMatchParameter": {
+      "name": "If-None-Match",
+      "in": "header",
+      "required": false,
+      "description": "Defines the If-None-Match condition. The operation will be performed only if the ETag on the server does not match this value.",
+      "type": "string",
+      "x-ms-parameter-grouping": {
+        "name": "access-condition"
+      },
+      "x-ms-parameter-location": "method"
+    },
+    "OperationIdParameter": {
+      "name": "operationId",
+      "in": "path",
+      "description": "Operation identifier.",
+      "minLength": 1,
+      "maxLength": 256,
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "LogCollectionIdParameter": {
+      "name": "operationId",
+      "in": "path",
+      "description": "Log collection identifier.",
+      "minLength": 1,
+      "maxLength": 256,
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "x-ms-client-name": "logCollectionId"
+    },
+    "DeviceIdParameter": {
+      "name": "deviceId",
+      "in": "path",
+      "description": "Device identifier in Azure IoT Hub.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "ModuleIdParameter": {
+      "name": "moduleId",
+      "in": "path",
+      "description": "Device module identifier in Azure IoT Hub.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "DeviceClassIdParameter": {
+      "name": "deviceClassId",
+      "in": "path",
+      "description": "Device class identifier.",
```

Comment:
```suggestion
      "description": "Device class identifier (combination of device manufacturer and model).",
```

Reply:
While `deviceClassId` is calculated from device manufacturer and model, it is not a clear concatenation of those. It is formatted hash value (byte[]) of various things (manufacturer and model is part of it).

Reply:
I'm imagining a user trying to get a device class for a known manufacturer/model. If the `deviceClassId` field is hashed and so not user-composable, can they use the `filter` field against `compatProperties` in during a `GET /deviceUpdate/{instanceId}/management/deviceClasses` to obtain the hashed deviceClassId from the result?

I ask because we explicitly mention in the property description for `filter` of that operation that `friendlyName` is filterable, but not compatProperties which is what I'd imagine I'd be after if I knew the manufacturer/model pair and needed the corresponding `deviceClassId`.

Reply:
I will add an explanation of the device class identifer/device class subgroup identifier to the DeviceClass and DeviceClassSubgroup object models that should hopefully explain this

Reply:
I will also clarify the device class description in ListDeviceClasses to be more accurate for this API version